### PR TITLE
TM Edge Clusters and QoS

### PR DESCRIPTION
**Pull request by Didainius**
This PR adds support for TM Edge Clusters and their QoS settings

### CODE REVIEW

File: govcd/tm_edge_cluster.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+const labelTmEdgeClusterSync = "TM Edge Cluster Sync"
+const labelTmEdgeClusterTransportNodeStatus = "TM Edge Cluster Transport Node Status"
+
+// TmEdgeCluster manages read operations for NSX-T Edge Clusters and their QoS settings
+type TmEdgeCluster struct {
+	TmEdgeCluster *types.TmEdgeCluster
+	vcdClient     *VCDClient
+}
+
+// wrap is a hidden helper that facilitates the usage of a generic CRUD function
```
Comment (lvirbalas) [inline govcd/tm_edge_cluster.go:20]: This is a bit confusing. By "hidden" you mean private? Do we really need this function? It seems so small and I can't seem to find its usage in this PR.

Comment (Didainius) [reply govcd/tm_edge_cluster.go:20]: It is everywhere, where we use generic methods. It is required to make generic approach work - https://github.com/vmware/go-vcloud-director/blob/main/govcd/openapi_generic_outer_entities.go#L13-L19

Comment (lvirbalas) [reply govcd/tm_edge_cluster.go:20]: Ah, truly. Thanks! Friday before holidays effect..

**Review by lvirbalas — APPROVED** [review]

**Review by adambarreiro — APPROVED** [review]
LGTM! Test passed

**Pull request merged**